### PR TITLE
Fix image reader direction

### LINKED ISSUE #691: GetDirection() behaves differently when called from ImageFileReader than from Image
When I use an ImageFileReader, call the ReadImageInformation() method then call GetDirection(), the tuple that is output is column-major whereas when I call GetDirection() from the Image itself (after reader.Execute()), the tuple is row-major.
So I have to transpose my 3x3 Direction matrix when I call direction from reader rather than the Image itself.
I don't need to load the images, I just need to get their information.
Is this normal behavior?

### PR DESCRIPTION
**Pull request by blowekamp**
Closes #691

### CODE REVIEW

File: Code/IO/include/sitkImageFileReader.h [hunk 1]
```diff
@@ -40,7 +40,10 @@ namespace itk {
      *
      * An interface is also provided to access the information from
      * the underlying itk::ImageIO. This information can be loaded
-     * with the ReadImageInformation method.
```
Comment (zivy) [inline Code/IO/include/sitkImageFileReader.h]: "The information is from the from" -> "The information is from the itk::ImageIO..."

"In some degenerate cases the reading the bulk data may produce difference results." -> 
"In some degenerate cases reading the bulk data may produce different results.
"

File: Testing/Unit/sitkImageIOTests.cxx [hunk 2]
```diff
@@ -563,23 +563,32 @@ TEST(IO, ImageFileReader_ImageInformation )
...
   EXPECT_EQ(reader.GetNumberOfComponents(), 1u);
+  EXPECT_EQ(reader.GetNumberOfComponents(), image.GetNumberOfComponentsPerPixel());
   EXPECT_VECTOR_DOUBLE_NEAR(reader.GetOrigin(), v3(-112, -21.687999, 126.894000 ), 1e-6);
+  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetOrigin(), image.GetOrigin(), 1e-6);
   EXPECT_VECTOR_DOUBLE_NEAR(reader.GetSpacing(), v3(0.859375, 0.8593899, 1.600000 ), 1e-6);
-  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetDirection(), v9(1, 0, 0,
-                                                      0, 0.46665081166793676, -0.88444164305490258,
-                                                      -0, 0.88444164305490258, 0.46665081166793676 ), 1e-8);
+  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetSpacing(), image.GetSpacing(), 1e-6);
+  EXPECT_VECTOR_DOUBLE_NEAR(reader.GetDirection(), v9(1, 0, -0,
```
Comment (zivy) [inline Testing/Unit/sitkImageIOTests.cxx]: Out of curiosity, does the sign of the zero (-0 vs. 0) matter here? I don't expect it to. Is this related to how EXPECT_VECTOR_DOUBLE_NEAR is implemented?

**Review by zivy — COMMENTED** [review]
Just some comment corrections and a minor question. The actual bug fix looks good to me.

File: Testing/Unit/sitkImageIOTests.cxx (hunk 2 above)
Comment (blowekamp) [reply Testing/Unit/sitkImageIOTests.cxx]: Technically IEEE 754 floating point representation has two zeros for the sign. This baseline value is now how it was printed. It makes now numeric difference, but visually indicate the proper matrix orientation.

**Comment by blowekamp** [comment]
Thanks for the review Ziv. I have address the issues. merging...

**Pull request merged**